pallets/parachain-staking/src/lib.rs

```diff
@@ -437,30 +437,82 @@ pub mod pallet {
 		}
 	}
 
+	#[derive(Clone, PartialEq, Encode, Decode, RuntimeDebug)]
+	pub enum NominatorStatus {
+		/// Active with no scheduled exit
+		Active,
+		/// Schedule exit to revoke all ongoing nominations
+		Leaving(RoundIndex),
+	}
+
 	#[derive(Encode, Decode, RuntimeDebug)]
+	/// Nominator state
+	pub struct Nominator2<AccountId, Balance> {
+		/// All current nominations
+		pub nominations: OrderedSet<Bond<AccountId, Balance>>,
```

Comment:
split this field into
collators: OrderedSet<AccountId>,
nominations: Vec<Bond<AccountId, Balance>>,

could be done in a follow up PR, would probably improve performance

Reply:
will do so in a follow up